src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1386,4 +1387,21 @@ public static void onAdvancement(EntityPlayerMP player, Advancement advancement)
     {
         MinecraftForge.EVENT_BUS.post(new AdvancementEvent(player, advancement));
     }
+    
+    // FORGE - Fix for MC-92916
+    public static void updatePlayersOnChunkUnload(World world, ClassInheritanceMultiMap<Entity>[] entityLists)
+    {
+        final List<EntityPlayer> players = new ArrayList<EntityPlayer>();
+        for (final ClassInheritanceMultiMap<Entity> multimap : entityLists)
+        {
+            for(final EntityPlayer player : multimap.getByClass(EntityPlayer.class))
```

Comment:
I think this loop can be condensed to:
```java
multimap.getByClass(EntityPlayer.class).forEach(players::add);
```

Similarly, the updating can be:
```java
players.forEach(player -> world.updateEntityWithOptionalForce(player, false));
```

Not sure if that makes it short enough to be patched inline, but it's worth a thought.

Reply:
So I guess the patch could become:

```java
// FORGE - Fix for MC-92916
final List<net.minecraft.entity.player.EntityPlayer> players = new java.util.ArrayList<net.minecraft.entity.player.EntityPlayer>();
java.util.Arrays.Arrays.stream(entityLists).forEach(multimap -> multimap.getByClass(net.minecraft.entity.player.EntityPlayer.class).forEach(players::add));
players.forEach(player -> world.updateEntityWithOptionalForce(player, false));
```

Would that be preferable to a new method in ForgeHooks? I'm more than happy to go with whatever the Forge maintainers think is best.

Reply:
1.12 Depends on Java8 anyway. Its fine to use them.